packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -347,6 +348,11 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
         const appTree = hierarchicalTree.trees[".app"];
         const protocolTree = hierarchicalTree.trees[".protocol"];
         if (appTree && protocolTree) {
+            if (this.hostPolicy.summarizerClient) {
+                // This returns a summary tree but we don't need it.
+                await this.buildPathToSummaryTreesCache(appTree, ".app");
```

Comment:
I do not know if it's right, but if possible, it would be great to work on final combined tree, and not expose the notion of ".app" and ".protocol" sub-trees as distinct entities to most of the code. I.e. can we call it once for the result  of combineProtocolAndAppSnapshotTree() call below?

Reply:
moved it into 1 function.